src/silo/preprocessing/preprocessor.cpp

```diff
@@ -126,6 +128,42 @@ Database Preprocessor::preprocess() {
    );
 }
 
+void Preprocessor::checkConfig() {
```

Comment:
I find this function name a little confusing

Reply:
Because of the name or because of the content? I'm also not super happy with the name. Suggestions are welcome ;)

Reply:
The name made me think that it would not mutate the config
Maybe finalize / resolve?

Otherwise an extra class ValidatedDatabaseConfig which also takes the ReferenceGenomes in its constructor would be viable

or we can leave it as a TODO for #426 , where we want to specify the unaligned sequences (and I would also say the aligned nuc and aa sequences) in the database config to reduce this dependency. Then the reference genomes is no longer a config for the sequences in the database but only a map for looking up the references

Reply:
Ok good point. `check` doesn't imply mutating. I renamed it to `finalizeConfig`.

`ValidatedDatabaseConfig` is only partially true, because you also need to validate it against the data (e.g. that all metadata fields are present etc.). So I'm not sure whether that would be a good name either.
If we want to redesign this anyway, then I'd propose to leave the rest to #426.